### PR TITLE
docs: update trg-8-03.md, introduce TruffleHog 

### PR DESCRIPTION
**Pull request by RoKrish14**
## Description

This PR introduces TruffleHog as a new open source tool for secret scanning to be used alongside native Github Secret scanning. This is being enforced as a replacement to the existing GitGuardian (commercial) tool.

Example workflow:
1. https://github.com/eclipse-tractusx/bpdm/blob/main/.github/workflows/trufflehog.yml
2. https://github.com/eclipse-tractusx/demand-capacity-mgmt/blob/main/.github/workflows/trufflehog.yml

Please note: The TRG checks continues for 24.08 under GitGuardian for secret scanning.
The teams who have already implemented TruffleHog, can also be considered for the TRG checks.

**NOTE: Please do not merge this PR until the August release**

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: docs/release/trg-8/trg-8-03.md
Comment (FaGru3n) [inline docs/release/trg-8/trg-8-03.md:44]: i would include the example as written in text directly into your example 😉 

> `schedule`: Schedule the workflow to run at least once a week with `0 0 * * 0`.

Comment (RoKrish14) [reply docs/release/trg-8/trg-8-03.md:46]: @FaGru3n Thanks, incorporated 👍

**Review by FaGru3n — APPROVED** [review]

**Comment by RoKrish14** [comment]
**NOTE: Please do not merge this PR until the August release**

**Comment by matbmoser** [comment]
@RoKrish14 you could have placed it as a draft then

**Comment by tom-rm-meyer-ISST** [comment]
@eclipse-tractusx/automotive-tractusx-committers please review

File: docs/release/trg-8/trg-8-03.md
Comment (tom-rm-meyer-ISST) [inline docs/release/trg-8/trg-8-03.md:9]: Please reformat table: 

```suggestion
| Status | Created     | Post-History                                                  |
|--------|-------------|---------------------------------------------------------------|
| Update | 27-Jun-2024 | Switching to TruffleHog due to GitGuardian licence expiration |
| Active | 26-Mar-2024 | Initial release                                               |
| Draft  | 04-Mar-2024 | Draft release                                                 |
```

**Review by tom-rm-meyer-ISST — APPROVED** [review]
Please reformat table (accept suggestion). Rest is great, thanks!

File: docs/release/trg-8/trg-8-03.md
Comment (ndr-brt) [inline docs/release/trg-8/trg-8-03.md:63]: this gives error when the PR is opened from a fork ([example](https://github.com/eclipse-tractusx/tractusx-edc/actions/runs/10299397391/job/28506589692?pr=1486)), it can be removed, as `checkout` automatically picks up the PR's branch

**Comment by evegufy** [comment]
@RoKrish14 are you still integrating review comments?

**Comment by danielmiehle** [comment]
@RoKrish14 Are you still working on this issue? I think we have a consensus on how we want to proceed here. Please get back by the end of next week, otherwise we would create a new PR for this topics. Thank you!

**Comment by RoKrish14** [comment]
@evegufy and @danielmiehle : Apologies for the delay as I was on vacation.
Making the necessary changes right away :)

File: docs/release/trg-8/trg-8-03.md
Comment (RoKrish14) [reply docs/release/trg-8/trg-8-03.md]: Hi @ndr-brt: Yes, you are right, thanks for notifying. I have made the changes.

**Review by ndr-brt — APPROVED** [review]

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks for the updates!

**Pull request merged**
